resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/config.scala

```diff
@@ -447,6 +447,13 @@ package object config extends Logging {
     .stringConf
     .createWithDefault("yarn.io/fpga")
 
+  private[spark] val APPLICATION_LIFETIME_TIMEOUT = ConfigBuilder("spark.yarn.app.lifetime.timeout")
```

Comment:
we are trying to not extend configs with more components (ie .xx.yy.)  I think spark.yarn.app.lifetimeTimeout should be sufficient.

Also you need to update docs/running-on-yarn.md